comms/dht/src/crypt.rs

```diff
@@ -105,29 +100,36 @@ pub fn create_origin_mac_challenge(header: &DhtMessageHeader, body: &[u8]) -> Ch
     )
 }
 
-/// Generates a challenge for the origin MAC.
-pub fn create_origin_mac_challenge_parts(
+/// Generates a 32-byte hashed challenge that commits to all message parts
+pub fn create_message_challenge_parts(
     protocol_version: DhtProtocolVersion,
     destination: &NodeDestination,
     message_type: DhtMessageType,
     flags: DhtMessageFlags,
     expires: Option<EpochTime>,
     ephemeral_public_key: Option<&CommsPublicKey>,
     body: &[u8],
-) -> Challenge {
+) -> [u8; 32] {
     let mut mac_challenge = Challenge::new();
-    mac_challenge.update(&protocol_version.to_bytes());
-    mac_challenge.update(destination.as_inner_bytes());
+    mac_challenge.update(&protocol_version.as_bytes());
+    mac_challenge.update(destination.to_inner_bytes());
```

Comment:
Is this guaranteed to have a fixed-length byte representation?

Reply:
Looks like this has a fixed-length representation, and types are differentiated using a flag.

Reply:
It is possible, if desired, to use variable-length representations. In this case, the length (as a fixed-length representation!) should be prepended after the (fixed-length) flag.

Reply:
That's right, each variant of `NodeDestination` has a flag in the first byte of the encoding. Gotcha, both the NodeId and CommsPublicKey have fixed-length representations so I don't think we'll have a need for variable-length encodings here.